File: Scripts/Para W7/Post/Verificador.cs
using System;
using System.Collections.Generic;
using System.IO;

namespace VerificadorDeInstalacion
{
    public static class Verificador
    {
        private static readonly Dictionary<string, string[]> PROGRAMS = new Dictionary<string, string[]>
        {
            { "GanaT_Bolivares", new string[] { @"C:\GanaT_Bolivares\GanaT.exe" } },
            { "GanaT_Pesos", new string[] { @"C:\GanaT_Pesos\GanaT.exe" } },
            { "GanaT_Dolares", new string[] { @"C:\GanaT_Dolares\GanaT.exe" } },
            { "GanaT", new string[] 
                { 
                    @"C:\%USERPROFILE%\Desktop\GanaT Bolivares.lnk",
                    @"C:\%USERPROFILE%\Desktop\GanaT Dolares.lnk",
                    @"C:\%USERPROFILE%\Desktop\GanaT Pesos.lnk"
                } 
            },
            { "Accesos_Directos", new string[]
                {
                    @"C:\%USERPROFILE%\Desktop\PagoListo.lnk",
                    @"C:\%USERPROFILE%\Desktop\Bemovil.lnk",
                    @"C:\%USERPROFILE%\Desktop\MisMarcadores.lnk",
                    @"C:\%USERPROFILE%\Desktop\SuperGana.lnk",
                    @"C:\%USERPROFILE%\Desktop\Payall.lnk",
                    @"C:\%USERPROFILE%\Desktop\Visitanos en Gana Loterias.lnk"
                }
            },
            { "Serial", new string[]
                {
                    @"C:\%USERPROFILE%\Desktop\KingDeportes.lnk",
                }
            }
        };

        public static List<string> VerificarProgramas(List<string> programas)
        {
            List<string> faltantes = new List<string>();
            foreach (string programa in programas)
            {
                if (PROGRAMS.ContainsKey(programa))
                {
                    bool encontrado = false;
                    foreach (string path in PROGRAMS[programa])
                    {
                        if (File.Exists(Environment.ExpandEnvironmentVariables(path)))
                        {
                            encontrado = true;
                            break;
                        }
                    }
                    if (!encontrado)
                    {
                        faltantes.Add(programa);
                    }
                }
            }
            return faltantes;
        }
    }
}

File: Scripts/Para W8 .1 y W11/Post/verificador.py
import os

PROGRAMS = {
    "aida64": {
        "path": r"C:\Program Files (x86)\FinalWire\AIDA64 Extreme\aida64.exe"
    },
    "rustdesk": {
        "path": r"C:\Program Files\RustDesk\rustdesk.exe"
    },
    "Anydesk": {
        "path": r"C:\Program Files (x86)\AnyDesk\Anydesk.exe"
    },
    "Xprinter": {
        "path": r"C:\XINYE POS Printer Driver\XPrinter Driver V7.77\XPrinter Driver V7.77.exe"
    },
    "winrar": {
        "path": r"C:\Program Files\WinRAR\WinRAR.exe"
    },
    "Crystaldisk": {
        "path": r"C:\Program Files\CrystalDiskInfo\DiskInfo64.exe"
    },
    "Edge": {
        "path": r"C:\Program Files (x86)\Microsoft\Edge\Application\msedge.exe"
    },
    "Chrome": {
        "path": r"C:\Program Files\Google\Chrome\Application\chrome.exe"
    },
    "GanaT_Bolivares": {
        "path": r"C:\GanaT_Bolivares\GanaT.exe"
    },
    "GanaT_Pesos": {
        "path": r"C:\GanaT_Pesos\GanaT.exe"
    },
    "GanaT_Dolares": {
        "path": r"C:\GanaT_Dolares\GanaT.exe"
    },
    "GanaT": {
        "paths": [
            r"C:\%USERPROFILE%\Desktop\GanaT Bolivares.lnk",
            r"C:\%USERPROFILE%\Desktop\GanaT Dolares.lnk",
            r"C:\%USERPROFILE%\Desktop\GanaT Pesos.lnk",
        ]
    },
    "Accesos_Directos": {
        "paths": [
            r"C:\%USERPROFILE%\Desktop\PagoListo.lnk",
            r"C:\%USERPROFILE%\Desktop\Bemovil.lnk",
            r"C:\%USERPROFILE%\Desktop\MisMarcadores.lnk",
            r"C:\%USERPROFILE%\Desktop\SuperGana.lnk",
            r"C:\%USERPROFILE%\Desktop\Payall.lnk",
            r"C:\%USERPROFILE%\Desktop\Visitanos en Gana Loterias.lnk"
        ]
    },
    "KingDeportes": {
        "paths": [
            r"C:\%USERPROFILE%\Desktop\KingDeportes.lnk",
        ]
    }
}

def verificar_programas(programas):
    faltantes = []
    for programa in programas:
        if programa in PROGRAMS:
            info = PROGRAMS[programa]
            if "paths" in info:
                for path in info["paths"]:
                    if not os.path.exists(os.path.expandvars(path)):
                        faltantes.append(programa)
                        break
            else:
                if not os.path.exists(info["path"]):
                    faltantes.append(programa)
    return faltantes
